### PR TITLE
refactor: Tasklist part of C8 single application

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

With this pull request, a Single Application (containing Tasklist, Operate and/or Zeebe) can be started via a `camunda.sh` script. By default, tasklist, operate and zeebe-broker profiles are applied.

Only basic authentication (`AUTH` profile) is supported.

## Related issues

relates to https://github.com/camunda/camunda/issues/18535

### CODE REVIEW

File: dist/pom.xml [hunk 1]
```diff
@@ -126,12 +126,6 @@
     <dependency>
       <groupId>io.camunda</groupId>
       <artifactId>operate-webapp</artifactId>
-      <exclusions>
```
Comment (houssain-barouni) [inline dist/pom.xml:129]: `commons-logging:commons-logging` is already excluded from `operate-common`

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/graphql/TasklistGraphQLConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import graphql.kickstart.execution.config.ObjectMapperProvider;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class TasklistGraphQLConfiguration {
+
+  @Bean
+  public ObjectMapperProvider objectMapperProvider(
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/graphql/TasklistGraphQLConfiguration.java:21]: to pass the right `ObjectMapper` bean

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/identity/IdentityAuthorizationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+@Component
+public class IdentityAuthorizationServiceImpl implements IdentityAuthorizationService {
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/identity/IdentityAuthorizationServiceImpl.java:30]: just extracting `IdentityAuthorizationService ` interface

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/tenant/TenantServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TenantServiceImpl implements TenantService {
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/security/tenant/TenantServiceImpl.java:23]: just extracting `TenantService` interface

**Comment by houssain-barouni** [comment]
@romansmirnov concerning the two points I had to fix in this pull request
- enable Tasklist security package when Operate is not enabled in the single application: checkout this commit https://github.com/camunda/camunda/pull/19139/commits/a7d9a7bc2f17ff5b96c9a27bcbf2afcff0057dda
- use same CSRF token header for Tasklist and Operate: made a separate pull request for that https://github.com/camunda/camunda/pull/19305

**Review by romansmirnov — APPROVED** [review]
Tested the PR together with CSRF token changes locally, the single application but also the standalone applications. LGTM!! Great stuff!

**Pull request merged**